### PR TITLE
Added subgraph methods to fetch markets by user position

### PR DESCRIPTION
**Pull request by gabrielfior**
Will be needed by microchain agent on PMA repo.

### CODE REVIEW

File: prediction_market_agent_tooling/markets/omen/omen_subgraph_handler.py [hunk 1]
```diff
@@ -329,6 +330,25 @@ def get_omen_market(self, market_id: HexAddress) -> OmenMarket:
 
         return omen_markets[0]
 
+    def get_omen_market_by_title(self, market_question_title: str) -> OmenMarket | None:
```
Comment (kongzii) [inline prediction_market_agent_tooling/markets/omen/omen_subgraph_handler.py:333]: Wdyt about adding `market_question_title` as filter to `get_omen_binary_markets`?

As you say, it's possible to have multiple markets with the same title, so seems like that method suits it well.

File: prediction_market_agent_tooling/markets/omen/omen_subgraph_handler.py [hunk 2]
```diff
@@ -505,3 +525,26 @@ def get_answers(self, question_id: HexBytes) -> list[RealityAnswer]:
         result = self.sg.query_json(fields)
         items = self._parse_items_from_json(result)
         return [RealityAnswer.model_validate(i) for i in items]
+
+    def get_markets_from_all_user_positions(
+        self, user_positions: list[OmenUserPosition]
+    ) -> list[OmenMarket]:
+        condition_ids: list[HexBytes] = sum(
```
Comment (kongzii) [inline prediction_market_agent_tooling/markets/omen/omen_subgraph_handler.py:532]: There can be potentially many duplicates

```suggestion
        condition_ids: list[HexBytes] = list(set(sum(
```

File: prediction_market_agent_tooling/markets/omen/omen_subgraph_handler.py [hunk 3]
```diff
@@ -505,3 +525,26 @@ def get_answers(self, question_id: HexBytes) -> list[RealityAnswer]:
...
+            [u.position.conditionIds for u in user_positions], []
+        )
```
Comment (kongzii) [inline prediction_market_agent_tooling/markets/omen/omen_subgraph_handler.py:534]:
```suggestion
        )))
```

(to make the suggestion above work)

File: prediction_market_agent_tooling/markets/omen/omen_subgraph_handler.py (hunk 1 above)
Comment (evangriffiths) [reply prediction_market_agent_tooling/markets/omen/omen_subgraph_handler.py:333]: An alternative suggestion: we could remove the need for this by having `GetMarkets` return the market id, and then we can use `subgraph_handler.get_omen_market_by_market_id` in the microchain agent's fuction directly

File: prediction_market_agent_tooling/markets/omen/omen_subgraph_handler.py (hunk 2 above)
Comment (gabrielfior) [reply prediction_market_agent_tooling/markets/omen/omen_subgraph_handler.py:513]: Great catch, yes for sure duplicates should be expected. Implemented.

File: prediction_market_agent_tooling/markets/omen/omen_subgraph_handler.py (hunk 1 above)
Comment (gabrielfior) [reply prediction_market_agent_tooling/markets/omen/omen_subgraph_handler.py]: I will go with @evangriffiths suggestion and reduce the amount of code necessary here. If we don't need to query by `market_title`, then no need for this method or added complexity.

File: tests/markets/omen/test_omen_subgraph_handler.py [hunk 4]
```diff
@@ -189,3 +193,58 @@ def test_get_user_positions_with_position_ids(
...
+        "0xffe4bf3e61be010728813a2a61ef422fd7d07b410170b64a5dfced9549f2e057"
+    )
+    positions = omen_subgraph_handler.get_positions(condition_id)
+    assert len(positions) == 2
+
+
+def build_incomplete_user_position_from_condition_ids(
+    condition_ids: list[HexBytes],
+) -> OmenUserPosition:
+    return OmenUserPosition.construct(
```
Comment (kongzii) [inline tests/markets/omen/test_omen_subgraph_handler.py:246]: Interesting one! Didn't know about it.

**Review by kongzii — APPROVED** [review]

File: tests/markets/omen/test_omen_subgraph_handler.py (hunk 4 above)
Comment (gabrielfior) [reply tests/markets/omen/test_omen_subgraph_handler.py:246]: It's a hacky way to bypass the model validation - use with care!

**Pull request merged**